docs/shaders.rst

```diff
@@ -0,0 +1,325 @@
+Writing PyGfx shaders
+=====================
+
+This document explains how to write shaders for PyGfx. This may be useful
```

Comment:
```suggestion
This document explains how to write shaders for PyGfx for the WgpuRenderer. 
This may be useful if you want to improve the existing shaders, add 
new shaders to PyGfx, or if you want to implement custom shaders in 
your own project.
```